cloudify_awssdk/common/decorators.py

```diff
@@ -93,11 +103,46 @@ def wrapper_inner(**kwargs):
                 ctx.instance.runtime_properties[key] = val
             # Add new operation arguments
             kwargs['resource_type'] = resource_type
-            kwargs['iface'] = class_decl(
-                ctx.node, logger=ctx.logger,
-                resource_id=utils.get_resource_id(
-                    node=ctx.node,
-                    instance=ctx.instance)) if class_decl else None
+
+            # Check if "aws_config" is provided
+            # if "client_config" is empty, then the current node is a swift
+            # node and the "aws_config" will be taken as "aws_config" for
+            # boto3 config in order to use the S3 API
+            aws_config = ctx.instance.runtime_properties.get('aws_config')
+            aws_config_kwargs = kwargs.get('aws_config')
+
+            # Attribute needed for AWS resource class
+            class_decl_attr = {
+                'ctx_node': ctx.node,
+                'logger': ctx.logger,
+                'resource_id': utils.get_resource_id(node=ctx.node,
+                                                     instance=ctx.instance),
+            }
+
+            # Check if "aws_config" is set and has a valid "dict" type because
+            #  the expected data type for "aws_config" must be "dict"
+            if aws_config:
+                if type(aws_config) is dict:
+                    class_decl_attr.update({'aws_config': aws_config})
+                else:
+                    # Raise an error if the provided "aws_config" is not a
+                    # valid dict data type
+                    raise NonRecoverableError('aws_config param must be '
+                                              'valid dict type ')
+
+            # Check the value of "aws_config" which could be part of "kwargs"
+            # and it has to be the same validation for the above "aws_config"
+            elif aws_config_kwargs:
+                if type(aws_config_kwargs) is dict:
+                    class_decl_attr.update({'aws_config': aws_config_kwargs})
+                else:
+                    # Raise an error if the provided "aws_config_kwargs"
+                    # is not a valid dict data type
+                    raise NonRecoverableError('aws_config param must be '
+                                              'valid dict type ')
```

Comment:
Better to print what type it actually is.

Reply:
Done